.codeclimate.yml

```diff
@@ -0,0 +1,43 @@
+version: "2"         # required to adjust maintainability checks
+checks:
+  # Methods or functions defined with a high number of arguments
+  argument-count:
+    config:
+      threshold: 6
+  # Boolean logic that may be hard to understand
+  complex-logic:
+    config:
+      threshold: 4
+  # Excessive lines of code within a single file
+  file-lines:
+    config:
+      threshold: 750
```

Comment:
I would set something like 120 or 160 and exclude files that violate this.

Reply:
120 seems fairly small. How about 250-500? I did some quick evaluation on the status-go repo to check on line numbers:

```
$ wc -l $(find . | grep "\.go" | grep -v vendor) | sort -nk1
     2 ./geth/jail/internal/main.go
     3 ./t/doc.go
     4 ./static/static.go
    10 ./geth/params/validator.go
    10 ./metrics/metrics_noop.go
    11 ./geth/api/utils.go
    11 ./metrics/node/utils.go
    12 ./geth/notifications/push/fcm/client.go
    12 ./geth/notifications/push/payload.go
    12 ./geth/signal/ios.go
    12 ./lib/main.go
    13 ./metrics/metrics_expvar.go
    16 ./lib/utils.go
    17 ./metrics/metrics_prometheus.go
    20 ./metrics/whisper/tracer_log.go
    21 ./geth/signal/signals_test.go
    22 ./geth/db/db.go
    22 ./geth/params/version.go
    22 ./geth/rpc/doc.go
    27 ./metrics/node/node_info_log.go
    28 ./geth/api/app_state_test.go
    28 ./geth/jail/internal/process/process.go
    30 ./profiling/heap.go
    30 ./services/shhext/signal.go
    31 ./geth/jail/internal/promise/promise.go
    31 ./geth/peers/discv5.go
    32 ./metrics/whisper/tracer_expvar.go
    33 ./geth/transactions/types_test.go
    33 ./t/destructive/testcontroller.go
    35 ./.gometalinter.json
    35 ./profiling/cpu.go
    36 ./geth/rpc/route_test.go
    36 ./metrics/node/node_info_prometheus.go
    37 ./metrics/node/node_info_expvar.go
    37 ./metrics/node/subscribe_test.go
    38 ./sign/request.go
    40 ./geth/transactions/addrlock.go
    40 ./profiling/profiler.go
    41 ./geth/transactions/fake/txservice.go
    41 ./sign/result.go
    42 ./geth/api/app_state.go
    44 ./cmd/statusd-cli/main.go
    48 ./geth/peers/cache_test.go
    48 ./t/e2e/rpc/client_test.go
    49 ./metrics/whisper/tracer_prometheus.go
    50 ./geth/node/node_api_test.go
    51 ./metrics/node/subscribe.go
    51 ./services/personal/service.go
    52 ./cmd/statusd/topics/topics.go
    52 ./extkeys/utils.go
    52 ./geth/jail/console/console.go
    52 ./sign/errors.go
    53 ./t/e2e/testing.go
    54 ./cmd/statusd/sync.go
    54 ./geth/jail/internal/timers/task.go
    54 ./geth/mailservice/mailservice.go
    54 ./geth/notifications/push/fcm/notification.go
    54 ./geth/peers/topic_register.go
    55 ./geth/params/cluster.go
    56 ./services/shhext/api.go
    58 ./cmd/statusd/whispercfg.go
    59 ./geth/api/connection_test.go
    61 ./geth/account/accounts_mock.go
    61 ./profiling/profiling_test.go
    62 ./geth/account/utils.go
    62 ./geth/peers/cache.go
    64 ./geth/api/connection.go
    64 ./t/e2e/whisper/mailservice_test.go
    67 ./geth/account/utils_test.go
    67 ./t/e2e/accounts/accounts_rpc_test.go
    71 ./geth/notifications/push/fcm/client_mock.go
    71 ./geth/notifications/push/fcm/notification_test.go
    77 ./lib/library_test.go
    81 ./geth/jail/doc.go
    86 ./geth/transactions/rpc_wrapper.go
    90 ./geth/transactions/fake/mock.go
    91 ./geth/transactions/types.go
    92 ./geth/rpc/route.go
    92 ./t/destructive/sync_test.go
    94 ./geth/jail/console/console_test.go
    95 ./cmd/statusd-cli/repl.go
   100 ./geth/jail/internal/promise/promise_test.go
   100 ./geth/jail/internal/vm/vm.go
   100 ./lib/types.go
   105 ./cmd/statusd/topics/topics_test.go
   105 ./geth/peers/topicpool_test.go
   108 ./geth/mailservice/api_test.go
   109 ./t/destructive/peers_test.go
   111 ./geth/signal/signals.go
   111 ./sign/notifications.go
   112 ./cmd/statusd/debug/debug.go
   112 ./geth/params/defaults.go
   116 ./geth/jail/internal/timers/timers.go
   117 ./geth/jail/internal/loop/loop_test.go
   121 ./services/personal/api.go
   124 ./extkeys/mnemonic_test.go
   128 ./geth/mailservice/api.go
   131 ./geth/node/status_node_rpc_client_test.go
   133 ./t/e2e/whisper/whisper_ext_test.go
   135 ./geth/jail/internal/loop/looptask/tasks.go
   140 ./geth/jail/internal/timers/timers_test.go
   146 ./t/e2e/suites.go
   148 ./geth/rpc/call_raw_test.go
   161 ./geth/peers/peerpool_test.go
   164 ./services/shhext/service.go
   177 ./cmd/statusd/debug/commands.go
   178 ./sign/pending_requests.go
   180 ./geth/jail/internal/fetch/fetch.go
   180 ./t/e2e/rpc/rpc_test.go
   183 ./geth/jail/handlers.go
   183 ./services/shhext/service_test.go
   188 ./geth/jail/handlers_test.go
   188 ./geth/rpc/client.go
   195 ./geth/jail/internal/loop/loop.go
   204 ./geth/transactions/transactor.go
   207 ./t/e2e/whisper/whisper_test.go
   216 ./cmd/statusd/debug/debug_test.go
   216 ./geth/rpc/call_raw.go
   218 ./geth/jail/cell.go
   219 ./geth/jail/jail_test.go
   222 ./geth/jail/internal/fetch/fetch_test.go
   223 ./geth/jail/cell_mock.go
   224 ./sign/pending_requests_test.go
   234 ./geth/peers/peerpool.go
   253 ./t/e2e/accounts/accounts_test.go
   255 ./geth/jail/cell_test.go
   260 ./geth/jail/internal/fetch/dist_fetch.go
   261 ./geth/api/api.go
   264 ./geth/peers/topicpool.go
   268 ./geth/node/node.go
   275 ./extkeys/mnemonic.go
   275 ./t/e2e/api/backend_test.go
   276 ./t/e2e/api/api_test.go
   287 ./t/e2e/jail/jail_test.go
   292 ./geth/node/status_node_test.go
   311 ./t/utils/utils.go
   317 ./t/e2e/services/personal_api_test.go
   318 ./geth/jail/internal/promise/js.go
   319 ./geth/jail/jail.go
   351 ./t/e2e/jail/jail_rpc_test.go
   358 ./cmd/statusd/main.go
   361 ./t/e2e/whisper/whisper_jail_test.go
   363 ./geth/api/backend.go
   364 ./geth/transactions/transactor_test.go
   379 ./t/e2e/node/manager_test.go
   389 ./geth/account/accounts.go
   398 ./extkeys/hdkey.go
   415 ./geth/account/accounts_test.go
   476 ./geth/node/status_node.go
   500 ./lib/library.go
   511 ./geth/params/config_test.go
   533 ./t/e2e/whisper/whisper_mailbox_test.go
   551 ./extkeys/hdkey_test.go
   611 ./geth/params/config.go
   661 ./static/bindata.go
   858 ./t/e2e/transactions/transactions_test.go
```

Reply:
Or we can even do 900 and a follow-up issue to bring it down to 500. There will be more of them anyway.